Python Flake8 fixes and Code refactoring

Nandan307 opened the pull request:
Improving code readability and optimizing it as per Python Flake8 convention

Nandan307 committed:
Python Flake8 fixes and Code refactoring

Improving code readability and optimizing it as per Python Flake8 convention

andrew103 commented on bot_template.py:
```diff
@@ -1,20 +1,19 @@
-irc = IRC()
-irc.connect(server, channel, nickname)
+
+irc_obj = IRC()
```
I would say lets keep the name of the instance to "irc" for the sake of consistency, simplicity, and syntactically more readable for function calls. Consistency in that the instance creation of the irc object in the class is set to "irc". Syntactically readable as in when calling a class method, `irc.send()` or `irc.connect()` reads better than `irc_obj.send()` or `irc.connect()` and so on.

Nandan307 replied:
I personally like to keep it easy on the eyes, since your import module is named "irc.py" and so is your instance, I thought naming the instance "irc_obj" would be easy on the eyes, but since this is your repository, I'll stick with the recommended instance name.

andrew103 commented on irc.py:
```diff
@@ -20,15 +19,20 @@ def change_nick(self, nick):
+        self.irc.connect((server, 6667))
+        # User authentication
+        self.irc.send("USER " + botnick + " " + botnick + " " +
+                      botnick + " :This is a fun bot!\n")
```
This could be simplified by rewriting the line as `self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")`. Could be worth checking out

Nandan307 replied:
To begin with,
>> self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")  //recommended
has 
>> botnick + " " + botnick + " " + botnick + " "
which is not the expected input as compared to botnick + " " + botnick + " " + botnick    //original
I gave it a try with 
>>  self.irc.send("USER " + (botnick + " ")*2 + botnick + ":This is a fun bot!\n")  

and this is what I got:
>> Traceback (most recent call last):
  File "spider_bot.py", line 57, in <module>
    spider_bot(irc, spider_bot)
  File "spider_bot.py", line 36, in spider_bot
    irc.connect(server, channel, nickname)
  File "/home/nakulkar/irc_bots/irc.py", line 27, in connect
    self.irc.send("USER " + (botnick + " ")*2 + botnick +
TypeError: unsupported operand type(s) for +: 'function' and 'str'

Nandan307 replied:
I gave it a try with 
$ self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")  
and it works!

andrew103 commented on thanos_bot.py:
```diff
@@ -1,26 +1,27 @@
-    if "PRIVMSG" in text and channel in text and "you should have gone for the head" in text.lower():
-        irc.send(channel, "*snap*")
+    if "PRIVMSG" in text and channel in text and\
+       "you should have gone for the head" in text.lower():
```
Since some of these conditional statements get pretty long, it could make sense to get a function that returns the result of the "PRIVMSG" and channel check. Could be called something like `check_msg()`

Nandan307 replied:
def check_msg(text):
    return "PRIVMSG" in text and channel in text

andrew103 replied:
You can indent if you use code blocks. Gotta use triple back-tick (like it is in markdown)

Nandan307 replied:
>def check_msg(text):
>> return "PRIVMSG" in text and channel in text

andrew103 commented on spider_bot.py:
```diff
@@ -1,46 +1,50 @@
+    if "PRIVMSG" in text and channel in text and ("tom brady" in text.lower()
+                                                  or "tb12" in text.lower() or
+                                                  "touchdown tommy" in
+                                                  text.lower()):
```
We could and should make this cleaner by structuring the conditionals in a more readable way. For example:
```
if conditional 
   and conditional
   and (conditional
       or conditional
       or conditional)
   and conditional:
    commands...
```
Not as extreme as this but within reason to make it more easily readable

Nandan307 replied:
Done

Nandan307 committed:
Emojibot: Lets the user use emojis on IRC

Emoji options compatible:
smiles, winks, laughs, fistbombs, heart, smirks, sleeps, cool, cries, tongue

NOTE: Use '*' before and after the option. Ex: *smiles*

Nandan307 committed:
Updated: Flake8 fixes and Code refactoring

Improving code readability and optimizing it as per Python Flake8 convention
Note: Suggested changes from Pull Request #1 have been considered.

andrew103 commented on spider_bot.py:
```diff
@@ -1,46 +1,59 @@
 
-    if "PRIVMSG" in text and channel in text and "idea" in text.lower():
-        irc.send(channel, "it's gonna be HUUUGGEEE")
+spider_bot(irc, nickname)
```
Any reason spider_bot doesn't have the `if __name__ == "__main___"` part here? Also, I don't think the irc instance needs to be passed in. It isn't passed in for thanos or emoji_bot.

Nandan307 replied:
re if __name__: I had accommodated your pastebin changes before your 180 turn. Forgot to change it back to my version.

re irc instance: In thanos_bot.py and emoji_bot.py, we don't use "irc.send()" and "irc.quit()" as we do in this file. Try testing it without passing in the instance as an argument and you'll get a compilation error.
